From 91f4812a8a71c5b09335913eed2ddfc60edd3da0 Mon Sep 17 00:00:00 2001
From: Martin Schmiedel <Martin.Schmiedel@tq-group.com>
Date: Thu, 21 May 2026 12:00:00 +0200
Subject: arm64: dts: freescale: enable symbols in TQ DTB's

Compile tq-boards with DTC -@ flag to generate __symbols__ node, which
is required for DT overlay application by U-Boot.

Signed-off-by: Martin Schmiedel <Martin.Schmiedel@tq-group.com>
---
 arch/arm64/boot/dts/freescale/Makefile | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 711e36cc2c99..32c6c11a8169 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -325,6 +325,12 @@ dtb-$(CONFIG_ARCH_MXC) += imx8mp-toradex-smarc-dev.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mp-tqma8mpql-mba8mpxl.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mp-tqma8mpql-mba8mp-ras314.dtb
 
+# TQ MBa8MPxL / RAS314: enable symbols node for DT overlay support
+DTC_FLAGS_imx8mp-tqma8mpql-mba8mpxl += -@
+DTC_FLAGS_imx8mp-tqma8mpql-mba8mp-ras314 += -@
+# TQ TQMa8MPxS / MB-SMARC-2: enable symbols node for DT overlay support
+DTC_FLAGS_imx8mp-tqma8mpqs-mb-smarc-2 += -@
+
 imx8mp-tx8p-ml81-moduline-display-106-av101hdt-a10-dtbs += imx8mp-tx8p-ml81-moduline-display-106.dtb \
 	imx8mp-tx8p-ml81-moduline-display-106-av101hdt-a10.dtbo
 imx8mp-tx8p-ml81-moduline-display-106-av123z7m-n17-dtbs += imx8mp-tx8p-ml81-moduline-display-106.dtb \
@@ -520,6 +526,10 @@ dtb-$(CONFIG_ARCH_MXC) += imx95-verdin-wifi-ivy.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx95-verdin-wifi-mallow.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx95-verdin-wifi-yavia.dtb
 
+# TQ TQMa9352: enable symbols node for DT overlay support
+DTC_FLAGS_imx93-tqma9352-mba93xxla-mini += -@
+DTC_FLAGS_imx93-tqma9352-mba93xxca += -@
+
 imx95-15x15-evk-pcie0-ep-dtbs = imx95-15x15-evk.dtb imx-pcie0-ep.dtbo
 dtb-$(CONFIG_ARCH_MXC) += imx95-15x15-evk-pcie0-ep.dtb
 imx95-19x19-evk-pcie0-ep-dtbs += imx95-19x19-evk.dtb imx-pcie0-ep.dtbo
-- 
2.43.0

